app/scripts/modules/core/src/domain/IManagedEntity.ts

```diff
@@ -24,6 +24,49 @@ export interface IManagedResourceSummary {
   };
 }
 
+export interface IManagedEnvironmentArtifact {
+  name: string;
+  type: string;
+  statuses: string[];
+  versions: {
+    current: string;
+    pending: string[];
+    approved: string[];
+    previous: string[];
+    vetoed: string[];
+  };
+}
+
+export interface IManagedEnviromentSummary {
+  artifacts: IManagedEnvironmentArtifact[];
+  name: string;
+  resources: string[];
+}
+
+export interface IManagedArtifactVersionEnvironmentSummary {
+  name: string;
+  state: string;
+  deployedAt?: string;
+  replacedAt?: string;
+  replacedBy?: string;
+}
+
+export interface IManagedArtifactVersion {
+  version: string;
+  environments: IManagedArtifactVersionEnvironmentSummary[];
```

Comment:
crazy idea: what if... we just didn't give this thing a name and inlined it? Seems somewhat futile to try to name a thing that's really just a particular join on a couple other discrete things.

Can always extract it like `IManagedArtifactVersion['environments']` if / when you need it on it's own.

Reply:
Yeah, I was thinking that too. Will go with that, at least for now.